code/datums/components/itempicky.dm

```diff
@@ -30,6 +34,9 @@
 
 /datum/component/itempicky/proc/particularly(datum/source, obj/item/pickingup)
 	SIGNAL_HANDLER
-	if(!is_type_in_typecache(pickingup, whitelist))
+	// if we were passed the output of a callback, check against that
+	// otherwise resolve our pointer
+	var/tertiary_result = tertiary_condition?.Invoke()
+	if(!tertiary_result && !is_type_in_typecache(pickingup, whitelist))
```

Comment:
should just do the callback directly (also change the comment cause e's outdated now)